src/routes/addons.ts

```diff
@@ -6,28 +6,34 @@
  * (Department of Information and Computing Sciences)
  */
 
-import { Request, Response } from "express";
 import { Context } from "../context";
-import { AddonCategory } from "prisma/prisma-client";
+import { Addon, AddonCategory } from "@prisma/client";
 import { join } from "node:path";
+import { z } from "zod";
 
 // TODO: move this to a better place
 const pageSize = 20;
 
-/**
- * The request type for getting addons.
- */
+////////////////////////////////////////////////////////////////////////////////
 
-interface GetAddonsRequest extends Request {
-  query: {
-    page: number;
-    category?: AddonCategory;
-  };
+interface GetAddonsRequest {
+  page: number;
+  category?: AddonCategory;
+}
+
+const getAddonsSchema = z.object({
+  page: z.coerce.number().int().gte(0).default(0),
+  category: z.nativeEnum(AddonCategory).optional()
+});
+
+interface GetAddonsResponse {
+  addons: Addon[];
 }
 
-export const getAddonsRoute =
-  (ctx: Context) => async (req: GetAddonsRequest, res: Response) => {
-    const { page, category } = req.query;
+export const getAddonsHandler =
+  (ctx: Context) =>
+  async (req: GetAddonsRequest): Promise<GetAddonsResponse> => {
+    const { page, category } = getAddonsSchema.parse(req);
```

Comment:
And `.coerce` parses the values for you, so need to get the values from `req` via zod's `parse` function.